src/components/EventDetailsPage.tsx

```diff
@@ -0,0 +1,49 @@
+import { ReactElement, useEffect, useState } from "react";
+import { useParams } from "react-router-dom";
+import { fetchEvent } from "../client/client";
+
+export type EventDetailsPageProps = {
+  event?: Event;
+};
+
+const EventDetailsPage = ({ event }: EventDetailsPageProps): ReactElement => {
+  const { id } = useParams();
+  const [loading, setLoading] = useState(false);
+  
+  let myEvent = event;
+
+  useEffect(() => {
+    setLoading(true);
+
+    fetchEvent(Number(id)).then().catch().finally();
```

Comment:
I think you can actually utilize `await` inside `useEffect` here instead, following this: https://ultimatecourses.com/blog/using-async-await-inside-react-use-effect-hook

Reply:
I think that would've been an issue since then it wouldn't render the page while it waits